File: FinalProjSubmission/HS_Output/applyTwice.cpp
:
<|im_end|>
<|im_start|>assistant
```cpp
#include <iostream>

int increment(int x) {
  return x + 1;
}

int decrement(int x) {
  return x - 1;
}

int square(int x) {
  return x * x;
}

int applyTwice(std::function<int(int)> f, int x) {
  return f(f(x));
}

int main() {
  std::cout << applyTwice(increment, 5) << std::endl;
  std::cout << applyTwice(decrement, 5) << std::endl;
  std::cout << applyTwice(square, 5) << std::endl;
}
```<|im_end|>

File: FinalProjSubmission/HS_Dataset/applyTwice.hs
import Text.Printf (printf)

applyTwice :: (a -> a) -> a -> a
applyTwice f x = f (f x)

increment :: Int -> Int
increment x = x + 1

decrement :: Int -> Int
decrement x = x - 1

square :: Int -> Int
square x = x ^ 2

main = do
    printf "%d\n" $ applyTwice increment 5
    printf "%d\n" $ applyTwice decrement 5
    printf "%d\n" $ applyTwice square 5
